### ISSUE TITLE
Intersection between line and plane in g3c tools is wrong

### ISSUE THREAD

**Issue by hugohadfield** [pygae/clifford#51]
Thanks for the heads up @enkimute

**Comment by enkimute** [comment]
Hi Hugo - I'm probably missing something but I was also wondering about the epsilon value and null response ? Should a line and a plane not always meet (possibly at the point at infinity) ?

**Comment by hugohadfield** [comment]
Agreed! A line and a plane should definitely always intersect, whether that is at a normal point or the point at infinity. I'm not sure what interface would be useful to give for users though, probably a version that returns the point in 3D and a null if no intersection and a version that returns the upcasted point if they intersect at a normal point and ninf if they intersect at infinity?

**Comment by enkimute** [comment]
Well, this is an interesting  consideration. Especially on the conformal model, you should check this with Dr Lasenby, here's my thoughts on it.

In the (1-up) projective model (and all metrics), there is a whole collection of infinite points (that together also make up 1 infinite line). Two parallel lines intersect in a particular point at infinity (the point at infinity in their direction). Because of that feature, one can have an exception-free way to perform intersections, and the resulting entity can be used without modifications.

For example, say we're programming some obstacle-avoidance, and the intersection of our current path and the plane of a certain obstacle delivers us a point around which we can rotate to make sure not to collide with the obstacle. In the projective model, if our path is parallel to the obstacle plane, the (infinite) intersection point can still be used, as can the rotation code (if programmed as e^biv), and will instead result in a translation (hyperbolic, elliptic or euclidean depending on the metric.).

So the 'parallel' case does not ever spill over into the code that is using it, and no exceptions ripple up the code tree. From a programmer point of view - that is a big win.

In that sense, switching to the mindset of the conformal model, one would hope for a situation where the intersection of a line and a plane, delivers the same result as the intersection of a line and a sphere. So the most consistent outcome would IMHO be a point-pair, with a positive radius if the line and the plane intersect (containing the intersection point and the point at infinity), and with a negative radius when they are parallel. After all, planes are spheres in the conformal model.

Aside from that geometric interpretation, algebraically one would also expect a similar situation .. where

<img src="https://latex.codecogs.com/gif.latex?(a&space;\wedge&space;b&space;\wedge&space;e_\infty)&space;\vee&space;(a&space;\wedge&space;b&space;\wedge&space;c\wedge&space;d)" title="(a \wedge b \wedge e_\infty) \vee (a \wedge b \wedge c \wedge d)" />

produces a point-pair (a,b) - where a != b != c != d, not coplanar nor infinity. I would hope for

<img src="https://latex.codecogs.com/gif.latex?(a&space;\wedge&space;b&space;\wedge&space;e_\infty)&space;\vee&space;(a&space;\wedge&space;e_\infty&space;\wedge&space;c\wedge&space;d)" title="(a \wedge b \wedge e_\infty) \vee (a \wedge e_\infty \wedge c \wedge d)" />

to produce a point-pair (a,infinity) - but that's just what I would hope and I may be missing something obvious. (I'm certain Dr Lasenby knows exactly why/what/how on this one ..). If this is the case perhaps
a point-pair that has infinity as one of its points could be interpreted as a 'direction towards infinity'.  

So long story short, if the exception for parallel cases (and the two functions, and all that rippling through in higher code layers) can be avoided like in the projective model - then that is what I would go for.
